lib/pbench/server/api/resources/query_apis/index_mappings.py

```diff
@@ -0,0 +1,84 @@
+from http import HTTPStatus
+from flask import jsonify
+from flask_restful import abort, Resource
+from logging import Logger
+from typing import Any, AnyStr, Dict
+
+from pbench.server import PbenchServerConfig
+from pbench.server.database.models.template import Template
+
+
+class IndexMappings(Resource):
+    """
+    Get properties of the run-data document type on which the dashboard can search.
+    This API currently only returns mapping properties of run-data documents.
+
+    TODO: In future we can tweak this API to take the document type name from the user
+    and query that type name against the template table to return the appropriate mappings
+
+    :return:
+    {
+        "@metadata":
+            [
+                "controller_dir",
+                "file-date",
+                "file-name",
+                "file-size",
+                "md5",
+                ...
+            ],
+       "host_tools_info":
+            [
+                "hostname",
+                "hostname-f",
+                ...
+            ],
+        "authorization":
+            [
+                "access",
+                "owner"
+            ]
+        "run":
+            [
+                "id",
+                "controller",
+                "user",
+                "name",
+                ...
+            ],
+        "sosreports":
+            [
+                "hostname-f",
+                "sosreport-error",
+                "hostname-s",
+                ...
+            ]
+    }
+    """
+
+    def __init__(self, config: PbenchServerConfig, logger: Logger):
+        self.config = config
+        self.logger = logger
+
+    def get(self) -> Dict[AnyStr, Any]:
+        """
+        Return mapping properties of the run-data document by querying the template database
+        """
+        try:
+            template = Template.find("run")
+            mappings = template.mappings
+
+            result = {}
+            for property in mappings["properties"]:
+                if "properties" in mappings["properties"][property]:
+                    result[property] = list(
+                        mappings["properties"][property]["properties"].keys()
+                    )
+
+            # construct response object
+            return jsonify(result)
+        except Exception:
+            self.logger.exception(
+                "Exception occurred while querying mappings for run documents."
+            )
+            abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR")
```

Comment:
If we do keep the `try` block here, do we really want to have just the one, catch-all case?  Or, are there specific cases that we should try to catch (like missing `"run"` document, or unable to connect to the database, et al.) which would ease debugging of issues?

Reply:
Yeah It applies to 68 only, and the reason I kept it catch-all and return 500 is that if we encounter any exception querying a database that's automatically 500 for an end user. But you are probably right that in this case we dont need a try block to catch 500 as it will be normal result of unhandled exception